pkg/system_config/syncer.go

```diff
@@ -0,0 +1,215 @@
+	}
+	for _, registry := range insecureRegistries {
+		rc := s.registriesConfContent.getRegistryConfOrCreate(registry)
+		rc.Insecure = &trueValue
+	}
+	s.ch <- true
+	return nil
+}
+
+func (s *SystemConfigSyncer) StoreRegistryCerts(registryCertTuples []registryCertTuple) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	s.registryCertTuples = registryCertTuples
+	s.ch <- true
+	return nil
+}
+
+func (s *SystemConfigSyncer) UpdateRegistryMirroringConfig(registry string, mirrors []string) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	rc := s.registriesConfContent.getRegistryConfOrCreate(registry)
+	rc.Mirrors = mirrors
+	s.ch <- true
+	return nil
+}
+
+func (s *SystemConfigSyncer) DeleteRegistryMirroringConfig(registry string) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	if rc, ok := s.registriesConfContent.getRegistryConf(registry); ok {
+		rc.Mirrors = []string{}
+		s.ch <- true
+		return nil
+	}
+	return fmt.Errorf("registry %s not found", registry)
+}
+
+func (s *SystemConfigSyncer) CleanupRegistryMirroringConfig() error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	for _, registry := range s.registriesConfContent.Registries {
+		registry.Mirrors = []string{}
+	}
+	s.ch <- true
+	return nil
+}
+
+func (s *SystemConfigSyncer) sync() error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	// marshall registries.conf and write to file
+	if err := s.registriesConfContent.writeToFile(); err != nil {
+		klog.Errorf("error writing registries.conf: %v", err)
+		return err
+	}
+	// marshall policy.json and write to file
+	if err := s.policyConfContent.writeToFile(); err != nil {
+		klog.Errorf("error writing policy.json: %v", err)
+		return err
+	}
+	// delete the certs.d content
+	if err := os.RemoveAll(DockerCertsDir); err != nil {
+		klog.Errorf("error deleting certs.d directory: %v", err)
+		return err
+	}
+	// write registry certs to file
+	for _, tuple := range s.registryCertTuples {
+		if err := tuple.writeToFile(); err != nil {
+			klog.Errorf("error writing registry cert: %v", err)
+			return err
+		}
+	}
+	return nil
+}
+
+// this should launch as a goroutine to consume events from the channel and write to disk
+func (s *SystemConfigSyncer) syncer() {
+	for {
+		select {
+		case <-s.ch:
+			if err := s.sync(); err != nil {
+				klog.Errorf("error syncing system config: %v", err)
+			}
+		}
+	}
+}
+
+// newSystemConfigSyncer creates a new SystemConfigSyncer object
+func newSystemConfigSyncer() IConfigSyncer {
+	ic := &SystemConfigSyncer{
+		registriesConfContent: defaultRegistriesConf(),
+		policyConfContent:     defaultPolicyConf(),
+		registryCertTuples:    []registryCertTuple{},
+		ch:                    make(chan bool),
+	}
+	go ic.syncer()
+	ctx := context.Background()
+	err := core.NewSingleObjectEventHandler[*v1.ConfigMap, *v1.ConfigMapList](ctx,
+		"image-registry-certificates", "openshift-image-registry",
+		time.Hour, func(et watch.EventType, cm *v1.ConfigMap) {
+			if et == watch.Deleted || et == watch.Bookmark {
```

Comment:
can these ever be deleted/absent? should we account for that

Reply:
Yep, thanks! Bad copy and paste from the other handler for the global pull secret. I'll add the methods already available in the syncer.

Reply:
mmm relooking at this, I think the events should not be handled by us as they will trigger a reconcile in other operators that will re-create the objects, hence triggering this handler again for the correct event to handle